dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -180,29 +191,47 @@ struct InstrumentationPoint<'tcx> {
     metadata: EventMetadata,
 }
 
-struct CollectFunctionInstrumentationPoints<'a, 'tcx: 'a> {
+struct InstrumentationAdder<'a, 'tcx: 'a> {
```

Comment:
NAK on this rename. This visitor collects a set of instrumentation points that we want to add to the program. It doesn't itself add instrumentation to the program, and the distinction between "adding" and "applying" instrumentation points is not clear to a reader.

Reply:
I do think it should be a noun not a verb, though.

Reply:
As verbose as it is, I like `CollectInstrumentationPointsVisitor`--mentioning that it is a visitor explains why the struct exists in the first place (to hang the Visitor impl and intermediate state off of), and the "CollectInstrumentationPoints" portion explains what its specific task is.